content/personal/telepathicgrunt/open-source-licenses.md

```diff
@@ -0,0 +1,51 @@
+
+- **OSL-3.0:** <https://opensource.org/license/osl-3-0-php> : <https://www.tldrlegal.com/license/open-software-licence-3-0>
+
+  - This one is more relaxed than LGPLv3 where copiers have to Open-Source their projects and give credit to the original code owner but does NOT need to add a link to the original repository nor state what changes were done. The copied code can be used in commercial Open-Source projects as well. A big difference with OSL-3.0 is that it closes the loophole where if a server owner modifies a serverside mod, they now must provide the source when asked.
+
+- **Apache-2.0:** <https://opensource.org/license/apache-2-0> : <https://www.tldrlegal.com/license/apache-license-2-0-apache-2-0>
+
+  - Modders taking your code must add a NOTICED file with attribution to you and state all the changes they did to the copied code. They do not need to Open-Source their mod and can use it in commercial ARR (All Rights Reserved) projects. It is similar to the MIT license but requires a more visible credit to the original owner of the copied code. Apache-2.0 also specifically allows “private use” so a server owner can modify your mod to use on a server without restriction.
+
+- **MPL-2.0:** <https://opensource.org/license/mpl-2-0> : <https://www.tldrlegal.com/license/mozilla-public-license-2-0-mpl-2>
+
+  - This one allows people to use your code in their commercial closed source projects. Just the copied and/or modified code portion must be publicly available or freely given when asked. This license also requires the modder to include a link to your repository so others can get the original software. They do not need to state any changes they made to your code. Server owners are allowed to modify the software as this license specifies that “distribution” is only about sending a copy of the software itself to another. See here for info about distribution: <https://www.mozilla.org/en-US/MPL/2.0/FAQ/#what-does-distribute-mean>
+
+> Notice, we did not mention GPL license. This is because GPL alone is not valid for a Minecraft mod because it requires all software linked to the mod to be Open-Source. And since Minecraft is essentially All Rights Reserved, that GPL license is not valid unless modified to have a classpath exception. We would encourage using LGPLv3 instead of GPL for ease if you were thinking of using GPL.
+
+Another thing to keep in mind, many of these licenses are only applicable to the code itself. For assets like textures or sound, you should put those under a different license. Modders tend to have a LICENSE.txt file for the code licenses and a LICENSE_ASSETS.txt file for asset licences. On CurseForge, they would pick the Custom License option and then put the two licenses there clarifying what each is applicable for.
```

Comment:
Perhaps it should state that the README of the repository should also state where each license applies, for clarity. (For example, stating the `LICENSE` applies to the whole repo except this one subdirectory or these types of files which are licensed udner `LICENSE-assets`).